### PR TITLE
Track role updates with on_member_update

### PR DESCRIPTION
**Pull request by teknogeek**

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -37,43 +42,39 @@ async def on_ready(self):
...
             return
-        # Only look in the active stream channel
-        channels_to_listen_to = {STREAM_CHAT_ID, WELCOME_CHAT_ID}
-        if message.channel.id not in channels_to_listen_to:
-            return
 
-        if message.channel.id == STREAM_CHAT_ID:
-            DB().accrue_channel_points(message.author.id, message.author.roles)
+        # Server Subscription message
+        if message.channel.id == WELCOME_CHAT_ID and message.type.value == 25:
```
Comment (braddotcoffee) [inline bot.py:51]: Not to be that guy but can we get a constant for this magic number

File: bot.py [hunk 2]
```diff
@@ -37,43 +42,39 @@ async def on_ready(self):
...
-                    break
-
-            if role_name is not None:
-                await self.get_channel(STREAM_CHAT_ID).send(
-                    f"Thank you {message.author.mention} for joining {role_name}!"
-                )
+            # comes in like this -> 'tier_name': 'THE ONES WHO KNOW membership',
+            # trim " membership" and get the actual role
+            role_name = role_sub_data.get("tier_name", "").rstrip(" membership")
+            role = next(filter(lambda x: x.name.startswith(role_name) and x.id in PREMIUM_IDS, message.guild.roles), None)
```
Comment (braddotcoffee) [inline bot.py:62]: This looks like you're trying to look up which role out of `PREMIUM_IDS` the member now has. Is there a reason we can't create a map of role name to ID?
```python
PREMIUM_IDS = {
    "THE ONES WHO KNOW": Config.CONFIG["Discord"]["Tier1RoleID"],
    ...
}
```

**Review by crscillitoe — APPROVED** [review]

File: bot.py (hunk 2 above)
Comment (teknogeek) [reply bot.py:62]: honestly just checking to see whether they are in the premium roles. I debated removing it entirely, wdyt

Comment (braddotcoffee) [reply bot.py:62]: We wouldn't get this message type if it's not a premium role, right?

Comment (teknogeek) [reply bot.py:62]: yeah, this is really to mitigate a scenario where there's another role starting with `THE ONES WHO KNOW` etc

Comment (braddotcoffee) [reply bot.py:65]: Just tell Christian not to make that role 🧠

**Pull request merged**